File: src/environment.c
#include "environment.h"
#include "raylib.h"

void draw_environment(){
    // Example: ground plane
    DrawPlane((Vector3){0.0f, 0.0f, 0.0f}, (Vector2){32.0f, 32.0f}, LIGHTGRAY);

    // Example: walls
    DrawCube((Vector3){-16.0f, 2.5f, 0.0f}, 1.0f, 5.0f, 32.0f, BLUE);
    DrawCube((Vector3){16.0f, 2.5f, 0.0f}, 1.0f, 5.0f, 32.0f, LIME);
    DrawCube((Vector3){0.0f, 2.5f, 16.0f}, 32.0f, 5.0f, 1.0f, GOLD);

    // If you need collision checks or environment logic, you might do it here
    // or in a separate "collision" or "physics" module.
}

void draw_player(Vector3 playerPosition) {
    // The player cube
    DrawCube(playerPosition, 0.5f, 0.5f, 0.5f, PURPLE);
    DrawCubeWires(playerPosition, 0.5f, 0.5f, 0.5f, DARKPURPLE);
}

File: client-rs/src/game/environment.rs
use color_eyre::owo_colors::OwoColorize;
use raylib::{
    color::Color,
    math::{Vector2, Vector3},
    prelude::RaylibDraw3D,
};

use super::player::Player;

pub fn draw_environment(d: &mut impl RaylibDraw3D) {
    // Ground
    d.draw_plane(
        Vector3::new(0.0, 0.0, 0.0),
        Vector2::new(32.0, 32.0),
        Color::LIGHTGRAY,
    );

    // Walls
    d.draw_cube(Vector3::new(-16.0, 2.5, 0.0), 1.0, 5.0, 32.0, Color::BLUE);
    d.draw_cube(Vector3::new(16.0, 2.5, 0.0), 1.0, 5.0, 32.0, Color::LIME);
    d.draw_cube(Vector3::new(0.0, 2.5, 16.0), 32.0, 5.0, 1.0, Color::GOLD);
    d.draw_cube(Vector3::new(0.0, 2.5, -16.0), 32.0, 5.0, 1.0, Color::ORANGE);
}
